### PR TITLE
Remove itemfreq

### PR DESCRIPTION
**Pull request by rolisz**
scipy.samplers.itemfreq has been deprecated since v1.1.0 and has been removed in 1.9.0.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
FYI: There is a recent stackoverflow question about this: [Issue with importing TextExplainer from eli5.lime package relating to deprecated 'itemfreq' function](https://stackoverflow.com/questions/73556675/issue-with-importing-textexplainer-from-eli5-lime-package-relating-to-deprecated)

**Review by b-mckenna — APPROVED** [review]

**Review by snowpedroferreira — APPROVED** [review]

**Review by lopuhin — APPROVED** [review]
Looks good, thanks @rolisz 👍 

EDIT: and sorry for a late review
